### ISSUE TITLE
Serializing optional with includingDefaultValueFields=true

### ISSUE THREAD

**Issue by C0mbatwombat** [curioswitch/protobuf-jackson#20]
Hi @chokoswitch,

Another question, I was looking into the behaviour of serializing `optional` fields with `includingDefaultValueFields=true`. It is the other way around from what I had expected looking at the values in the proto object: `non-optional` fields are serialized, while `optional` fields are not serialized.

Test case:
```
message TestProto {
  optional uint64 v1 = 1;
  uint64 v2 = 2;

  optional uint32 v3 = 3;
  uint32 v4 = 4;

  optional sint64 v5 = 5;
  sint64 v6 = 6;

  optional sint32 v7 = 7;
  sint32 v8 = 8;

  optional double v9 = 9;
  double v10 = 10;

  optional float v11 = 11;
  float v12 = 12;
}


TestProto testObject = TestProto.newBuilder().setV1(0).setV2(0).setV3(0).setV4(0).setV5(0).setV6(0).setV7(0).setV8(0).setV9(0.0).setV10(0.0).setV11((float) 0.0).setV12((float) 0.0).build();
```

will serialize to:
```
{
  "v2" : "0",
  "v4" : 0,
  "v6" : "0",
  "v8" : 0,
  "v10" : 0.0,
  "v12" : 0.0
}
```

This is the opposite from the values stored in the proto object itself:
```
testObject.toString():
v1: 0
v3: 0
v5: 0
v7: 0
v9: 0.0
v11: 0.0
```

Would it be possible to add a flag to change the behaviour so the object would serialize as:
```
{
  "v1" : "0",
  "v3" : 0,
  "v5" : "0",
  "v7" : 0,
  "v9" : 0.0,
  "v11" : 0.0
}
```

Thanks!

**Comment by chokoswitch** [comment]
Hi @C0mbatwombat - hmm, this seems like it may be a bug. The non-optional values should be printed because they are the default values, and the optional values should be printed because they are set. I would expect this output

```
{
  "v1" : "0",
  "v2" : "0",
  "v3" : 0,
  "v4" : 0,
  "v5" : "0",
  "v6" : "0",
  "v7" : 0,
  "v8" : 0,
  "v9" : 0.0,
  "v10" : 0.0,
  "v11" : 0.0
  "v12" : 0.0
}
```

We wouldn't expect your last output example unless `includeDefaultValues` isn't set, because that example is not outputing the default values.

Will take a look

**Comment by chokoswitch** [comment]
@C0mbatwombat I have pushed a fix, would you be able to confirm it works for you with the latest snapshot? To get the output you want in the last example, don't forget you will need `includingDefaultValueFields=false`, not `true`.

**Comment by chokoswitch** [comment]
Thanks for confirming @C0mbatwombat - I have released 2.3.0 with the fix.
